src/ejabberd_auth.erl

```diff
@@ -101,6 +111,78 @@ plain_password_required(Server) ->
               M:plain_password_required()
       end, auth_modules(Server)).
 
+-spec set_generic_opts(Host :: ejabberd:server(),
+                      Opt :: atom(),
+                      KVs :: [tuple()]) ->  {atomic|aborted, _}.
+set_generic_opts(Host, Opt, KVs) ->
+    OldOpts = ejabberd_config:get_local_option(Opt, Host),
```

Comment:
`set_generic_opts` should access `auth_opts` by default, so the API here should be `set_opt(Host, Opt, Val)` (including the function name change).